src/ScreenText.cpp

```diff
@@ -0,0 +1,144 @@
+        MessageBox(nullptr, TEXT("FW1CreateFactory"), TEXT("Error"), MB_OK);
+        assert(false);
+    }
+
+    HR = DWriteCreateFactory(DWRITE_FACTORY_TYPE_SHARED, __uuidof(IDWriteFactory), reinterpret_cast<IUnknown**>(&m_d_write_factory));
+
+    HR = m_d_write_factory->CreateTextFormat(
+        L"Arial",                        // Font family name
+        nullptr,                         // Font collection (NULL for the system font collection)
+        DWRITE_FONT_WEIGHT_NORMAL,
+        DWRITE_FONT_STYLE_NORMAL,
+        DWRITE_FONT_STRETCH_NORMAL,
+        m_font_size,                     // Font size
+        L"en-US",                        // Locale
+        &m_d_write_text_format
+    );
+
+    HR = m_d_write_factory->CreateTextLayout(m_text, wcslen(m_text), m_d_write_text_format, 2048, 2048, &m_d_write_text_layout);
+
+    DWRITE_TEXT_METRICS textMetrics;
+    if (SUCCEEDED(HR)) {
+        HR = m_d_write_text_layout->GetMetrics(&textMetrics);
+    }
+
+    if (SUCCEEDED(HR)) {
+        HR = m_d_write_factory->CreateTextLayout(
+            m_text,                    // Text to be laid out
+            wcslen(m_text),            // Length of the text
+            m_d_write_text_format,     // Text format
+            textMetrics.width,         // Use measured text width
+            textMetrics.height,        // Use measured text height
+            &m_d_write_text_layout     // Output text layout object
+        );
+    }
+
+    DWRITE_TEXT_RANGE tr;
+    tr.length = 256;
+    tr.startPosition = 0;
+
+    m_d_write_text_layout->SetFontSize(m_font_size, tr);
+    m_d_write_text_layout->SetTextAlignment(DWRITE_TEXT_ALIGNMENT_CENTER);
+    m_d_write_text_layout->SetParagraphAlignment(DWRITE_PARAGRAPH_ALIGNMENT_CENTER);
+
+    HR = m_FW1_factory->CreateFontWrapper(m_renderer->get_device(), L"Arial", &m_font_wrapper);
+    if (FAILED(HR)) {
+        MessageBox(nullptr, TEXT("CreateFontWrapper"), TEXT("Error"), MB_OK);
+        assert(false);
+    }
+
+    m_FW1_factory->Release();
+}
+
+void ScreenText::draw() const
+{
+    m_font_wrapper->DrawTextLayout(
+        m_renderer->get_device_context(),
+        m_d_write_text_layout,
+        m_position.x,
+        m_position.y,
+        m_color,
+        m_flags
+    );
+
+}
+
+std::string ScreenText::get_name() const
+{
+    std::string const name = typeid(decltype(*this)).name();
+    return name.substr(6);
+}
+
+void ScreenText::set_text(const WCHAR* new_content)
+{
+    // Update the text content
+    wcscpy_s(m_text, new_content);
+
+    // Update the DirectWrite text layout if necessary
+    if (m_d_write_text_layout != nullptr)
+    {
+        // Release the existing text layout
+        m_d_write_text_layout->Release();
+        m_d_write_text_layout = nullptr;
+    }
+
+    // Create a new text layout with the updated content
+    HRESULT HR = m_d_write_factory->CreateTextLayout(new_content, wcslen(new_content), m_d_write_text_format, 2048, 2048, &m_d_write_text_layout);
```

Comment:
We usually name this value lower-cased hr (or h_result in one file, but this will probably be renamed to hr as well)